network/src/noise/handshake.rs

```diff
@@ -249,49 +243,45 @@ impl NoiseUpgrader {
                 Some(&payload),
                 &mut client_noise_msg,
             )
-            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+            .map_err(NoiseHandshakeError::BuildClientHandshakeMessageFailed)?;
 
         // send the first handshake message
-        debug!(
-            NetworkSchema::new(&self.network_context),
-            noise_role = "client",
-            step = "writing_data",
-            remote_public_key = remote_public_key,
-            "{} noise client writing_data: {}",
+        trace!(
+            "{} noise client: handshake write: remote_public_key: {}",
             self.network_context,
-            remote_public_key
+            remote_public_key,
         );
-        socket.write_all(&client_message).await?;
-        socket.flush().await?;
+        socket
+            .write_all(&client_message)
+            .await
+            .map_err(NoiseHandshakeError::ClientWriteFailed)?;
```

Comment:
crazy idea, have a `NoiseHandshakeError::IO(std::io::Error)` type for these

Reply:
I want a more specific message, especially for the ClientRead case, which is like 90% of our errors. for the server errors, the read doesn't know the client peerid yet while the write and flush do